src/containers/AppBody.jsx

```diff
@@ -87,7 +87,7 @@ export class AppBody extends Component {
         }
       }, () => {
         // Failed to get the snippet, either bad URL or unauthorized
-        router.push('/');
+        router.push('/404');
```

Comment:
I don't think the url should change. We should just show the error message a la https://github.com/dane-johnson/bullshit-repo